### PR TITLE
PWN-9001 - Swap refresh routes interval via feature toggle

### PR DESCRIPTION
**Pull request by 1sakov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-9001

## Description of Work

[ what work was done in this ticket ]

## Screenshots (Optional)

| Screenshot #1 | Screenshot #2 | Screenshot #3 |
|:-------------:|:-------------:|:-------------:|
| [Paste Pic 1] | [Paste Pic 2] | [Paste Pic 3] |

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/common/feature_toggles/toggles/remote/RemoteFeatureToggle.kt [hunk 1]
```diff
@@ -39,6 +39,16 @@ abstract class IntFeatureToggle(
         }
 }
 
+abstract class LongFeatureToggle(
+    private val valuesProvider: RemoteConfigValuesProvider
+) : RemoteFeatureToggle<Long>() {
+    override val value: Long
+        get() = valuesProvider.getLong(featureKey) ?: kotlin.run {
+            Timber.tag("IntFeatureToggle").i("No value found for $featureKey; using defaults = $defaultValue")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/common/feature_toggles/toggles/remote/RemoteFeatureToggle.kt:47]: IntFeatureToggle -> LongFeatureToggle

File: app/src/main/java/org/p2p/wallet/common/feature_toggles/toggles/remote/SwapRoutesRefreshFeatureToggle.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import org.p2p.wallet.common.feature_toggles.remote_config.RemoteConfigValuesProvider
+
+private const val DEFAULT_INTERVAL_IN_MILLIS = 20_000L
+
+class SwapRoutesRefreshFeatureToggle(
+    valuesProvider: RemoteConfigValuesProvider
+) : LongFeatureToggle(valuesProvider) {
+    override val featureKey: String = "swap_route_refresh"
+    override val featureDescription: String = "The interval for refreshing routes"
+    override val defaultValue: Long = DEFAULT_INTERVAL_IN_MILLIS
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/common/feature_toggles/toggles/remote/SwapRoutesRefreshFeatureToggle.kt:12]: what if you add here valueDuration: Duration and use it like 
val valueDuration:Duration = value.milliseconds

so we won't guess what time unit is inside the actual value

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/common/feature_toggles/remote_config/AppFirebaseRemoteConfig.kt [hunk 3]
```diff
@@ -106,6 +106,13 @@ class AppFirebaseRemoteConfig(
         return remoteConfig.getString(toggleKey).toIntOrNull()
     }
 
+    override fun getLong(toggleKey: String): Long? {
```
Comment (eduardmaximovich) [inline app/src/main/java/org/p2p/wallet/common/feature_toggles/remote_config/AppFirebaseRemoteConfig.kt:109]: Perhaps for clarity, all these `get*` functions should be separated into two distinct ones:
1. `getLongOrNull(): Long?`
2. `getLong(key, default: Long): Long`

**Review by eduardmaximovich — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/common/feature_toggles/remote_config/AppFirebaseRemoteConfig.kt (hunk 3 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/common/feature_toggles/remote_config/AppFirebaseRemoteConfig.kt:109]: it's a inside impl of FirebaseRemoteConfig
we pass default values inside the feature toggles, so i think there no need for multiple methods

**Pull request merged**